### PR TITLE
New image picker dialog and composite presenter

### PR DESCRIPTION
**Pull request by SithEngineer**
This PR has many refactored code regarding the views of manage user and manage store to fix the issues:
https://aptoide.atlassian.net/browse/AN-1625
https://aptoide.atlassian.net/browse/AN-1685
https://aptoide.atlassian.net/browse/AN-1724

And for this fixes a new mechanism to ask for gallery and camera pictures is being used via a new image picker dialog. This new image picker dialog has a specific presenter. To use this presenter with the presenters from the views ManageUser and ManageStore a CompositePresenter was created.

### CODE REVIEW

**Comment by SithEngineer** [comment]
This PR is expected to also fix this issue https://aptoide.atlassian.net/browse/AN-1731

File: v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/PhotoFileGenerator.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  private final File storageDirectory;
+  private final String fileProviderAuthority;
+
+  public PhotoFileGenerator(Context context, File storageDirectory, String fileProviderAuthority) {
+    this.context = context;
+    this.storageDirectory = storageDirectory;
+    this.fileProviderAuthority = fileProviderAuthority;
+  }
+
+  //public Single<String> generateNewImageFileUriAsString() {
```
Comment (Trinkes) [inline v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/PhotoFileGenerator.java]: remove commented code

File: v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/store/ManageStoreFragment.java [hunk 2]
```diff
@@ -111,20 +195,71 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
...
+  @Override public void onDestroyView() {
+    dismissWaitProgressBar();
+    if (dialogFragment != null) {
+      dialogFragment.dismiss();
+      dialogFragment = null;
+    }
+    super.onDestroyView();
+  }
+
+  public void setupViews() {
```
Comment (Trinkes) [inline v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/store/ManageStoreFragment.java]: should this be called by presenter?

File: v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/store/ManageStoreNavigator.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  private final FragmentNavigator fragmentNavigator;
+  private final boolean goBackToHome;
+
+  public ManageStoreNavigator(FragmentNavigator fragmentNavigator, boolean goBackToHome) {
+    this.fragmentNavigator = fragmentNavigator;
+    this.goBackToHome = goBackToHome;
+  }
+
+  public void navigate() {
+    if (goBackToHome) {
```
Comment (Trinkes) [inline v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/store/ManageStoreNavigator.java]: should this be decided by navigator?

File: v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/user/ManageUserNavigator.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  public void back() {
+    navigator.popBackStack();
+  }
+
+  public void navigateAway(boolean isEditProfile) {
+    final boolean showPrivacyConfigs = Application.getConfiguration()
+        .isCreateStoreAndSetUserPrivacyAvailable();
+    if (isEditProfile) {
```
Comment (Trinkes) [inline v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/user/ManageUserNavigator.java]: should this logic be here?

File: v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/ImageLoaderFragment.java [hunk 5]
```diff
@@ -5,39 +5,12 @@
 
 package cm.aptoide.pt.v8engine.view.account;
```
Comment (SithEngineer) [inline v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/ImageLoaderFragment.java]: Removed.

File: v8engine/src/main/java/cm/aptoide/pt/v8engine/V8Engine.java [hunk 6]
```diff
@@ -433,6 +434,7 @@ public NotificationCenter getNotificationCenter() {
 
   public StoreManager getStoreManager() {
     if (storeManager == null) {
+      ImageLoader imageLoader = ImageLoader.with(getApplicationContext());
```
Comment (SithEngineer) [inline v8engine/src/main/java/cm/aptoide/pt/v8engine/V8Engine.java]: Removed this line of code.

File: v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/PhotoFileGenerator.java (hunk 1 above)
Comment (SithEngineer) [inline v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/PhotoFileGenerator.java]: Deleted commented code.

File: v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/store/ManageStoreFragment.java (hunk 2 above)
Comment (SithEngineer) [inline v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/store/ManageStoreFragment.java]: No. The view has its initial state, which in that case is set by the setupViews() method.

File: v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/store/ManageStoreNavigator.java (hunk 3 above)
Comment (SithEngineer) [inline v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/store/ManageStoreNavigator.java]: You're right. This decision should be done in the presenter.

File: v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/user/ManageUserNavigator.java (hunk 4 above)
Comment (SithEngineer) [inline v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/user/ManageUserNavigator.java]: No. This should be in the done in the presenter.

**Pull request merged**